File: Angular + SpringMVC/jiahua team/firstProject/frontend-angular/src/app/models/Item.ts
import { BaseModel } from './BaseModel';

export interface Item extends BaseModel {
  category: string;
  subCategory: string;
  verified: Date;
  imgUrl: string;

  // Product Summary
  description: { [key: string]: string };
  type$: { [key: string]: string };
  technical_specifications$: { [key: string]: string };

  // Product Details
  series_information: string[];

  // Contact
  sales_representative: { [key: string]: string };
  manufacturer: { [key: string]: string };

  // Past Selections
  past_selections$: { [key: string]: string };
}

File: Angular + SpringMVC/jiahua team/firstProject/marketplace/src/main/java/com/itlize/marketplace/entities/Item.java
package com.itlize.marketplace.entities;

import java.time.LocalDate;
import java.util.Map;
import java.util.Set;
import javax.persistence.ElementCollection;
import javax.persistence.Entity;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Data
@EqualsAndHashCode(callSuper = false)
public class Item extends BaseEntity {
  private static final long serialVersionUID = 817979112435251527L;

  private String category;
  private String subCategory;
  private LocalDate verified;
  private String imgUrl;

  // Product Summary
  private @ElementCollection Map<String, String> description;
  private @ElementCollection Map<String, String> type$;
  private @ElementCollection Map<String, String> technical_specifications$;

  // Product Details
  private @ElementCollection Set<String> series_information;

  // Contact
  private @ElementCollection Map<String, String> sales_representative;
  private @ElementCollection Map<String, String> manufacturer;

  // Past Selections
  private @ElementCollection Map<String, String> past_selections$;
}
